### PR TITLE
Close #37

### LINKED ISSUE #37: Enforce explicit dtypes throughout all DataFrame-based classes
Right now, both `TrecRun` and `TrecQrel` have whatever datatypes pandas automatically infers for each column. 

https://github.com/joaopalotti/trectools/blob/5c1d56e9cf955f45b5a1780ee6a82744d31e7a79/trectools/trec_run.py#L43

This leads to inconsistencies across different experiments based on the `doc_id` and `query` names used: some get inferred as `int64` (for example MSMARCO-Passage), some as `str` (for example ClueWeb12). While this is not a problem when working in the "intended" file-based way of loading and evaluating runs/qrels, it incurs some frustration when accessing the `TrecRun.run_data` and `TrecQrel.qrels_data` fields directly in in-memory experiments (related issue: https://github.com/joaopalotti/trectools/issues/13). 

Here, the experiment code has to be customized to the inferred pandas-dtype for each collection. For example, when an in-memory run is to be evaluated with qrels loaded from disk, one has to make sure that the custom `run.run_data` matches the dtypes of the qrels, which may change based on the collection used. In my case, I assumed `id` fields to be string-based and ran into a lot of merge-on-different-dtypes errors.

Therefore, I would argue towards enforcing a consistent dtype (`str`) when loading runs/qrels from disk for the `doc_id` and `query` columns. This would not affect the file-based workflow in any way but greatly improve the memory-based experience and usability by providing a consistent type mapping for all collections.

### PR DESCRIPTION
**Pull request by lgienapp**
This PR introduces changes to the way run- and qrel-files are loaded from disk/memory (#37). The goal is to (1) have forced typing and column name validation with default arguments; and (2) be able to circumvent both automatic checks if needed with custom arguments.

###### Flexible column naming
Tl;dr: `read` methods now all have a `header` argument that allows to specify custom column names. `load` methods have a `mapping` argument to optionally map external columns to the internal representation.

- replace `qrel_header` argument in `TrecQrel.read_qrel` with immutable default
- add `run_header` argument in `TrecRun.read_run` to mirror the `TrecQrel.read_qrel` signature
- add `column_mapping` argument to `TrecRun.load_run_from_dataframe` to optionally map columns to internal representation correctly

###### Forced Typing & Validation
Tl;dr: Columns with default names have forced types. This can be circumvented by using the `header`/`mapping` argument during data loading, as custom column names are not subject to typechecking.

- add forced string typing to `query` column in `TrecQrel`
- add forced string typing to `query`, `q0`, and `docid` column in `TrecRun`
- add `validate` argument to  `TrecRun.load_run_from_dataframe` to trigger column name validation (default True)
- column validation in `TrecRun.load_run_from_dataframe` now raises a ValueError instead of printing, and produces the full list of missing columns

###### Unittests
Tl;dr: unittest adaptation
- adapt `testtrecrun.py` unittests to forced string type
- adapt `testtreceval.py` unittests to forced string type
- remove print statement from `testtreceval.test_get_precision`

### CODE REVIEW

**Comment by joaopalotti** [comment]
Looks awesome, @lgienapp! Thanks for your PR!
